src/libraries/System.Private.Uri/src/System/Uri.cs

```diff
@@ -3530,25 +3525,13 @@ private unsafe void ParseRemaining()
             // Note that we don't support one-letter schemes that will be put into a DOS path bucket
 
             int end = idx;
-            while (end < length && uriString[end] != ':')
+            while (end < uriString.Length && uriString[end] != ':')
             {
                 ++end;
             }
 
-            // NB: On 64-bits we will use less optimized code from CheckSchemeSyntax()
-            //
-            if (IntPtr.Size == 4)
-            {
-                // long = 4chars: The minimal size of a known scheme is 2 + ':'
-                if (end != length && end >= idx + 2 &&
-                    CheckKnownSchemes((long*)(uriString + idx), end - idx, ref syntax))
-                {
-                    return end + 1;
-                }
-            }
```

Comment:
Why was this removed?  Did you benchmark on 32-bit as well and determine this wasn't helping?

Reply:
I removed it because its behavior is wrong on big-endian.
I ran the benchmark for 32bit and there is a ~4% regression in the worst-case.

If we believe this kind of optimization is necessary, I can add an alternative version of it - initial testing showed it can save ~15% from ctor time for simple Uris.

Reply:
> I removed it because its behavior is wrong on big-endian. I ran the benchmark for 32bit and there is a ~4% regression in the worst-case.

Rather than deleting it, can you instead just make it:
```C#
if (IntPtr.Size == 4 && BitConverter.IsLittleEndian)
```
?